arbnode/sync_monitor.go

```diff
@@ -69,8 +69,9 @@ func (s *SyncMonitor) SyncProgressMap() map[string]interface{} {
 		syncing = true
 	} else {
 		res["blockNum"] = lastBlockNum
+		res["messageOfLastBlock"] = lastBuiltMessage
+		arbutil.UpdateSequenceNumberGauge(lastBuiltMessage)
```

Comment:
I'm not sure this is called very consistently in the node (for most nodes I think it's only when you call `eth_syncing`). Also, I'm not sure this is what the gauge is measuring. The last block's message might be behind the last message in the database if block creation is falling behind. I'd consider removing this update to the gauge. Perhaps instead, on startup we could set the gauge to the current tx streamer message count?

Reply:
Ah, I just realized the relay update code only happens for relays not all broadcast clients. Perhaps instead, we could put it in `addMessages` in the tx streamer? (I might be misremembering the method name)

Reply:
Note: I've edited that last message a few times, so you might want to refresh if you read the original version, sorry

Reply:
I was thinking that it would be nice to have a metric of what the last block that finished executing was.  That way it would be easy to tell where in the chain the validator was, as well as if any of the secondaries were falling behind, etc.

Reply:
Refactored using setMessageCount.  Tsahi recommended adding a metric when sequence number is stored in block as well.